test/Microsoft.ML.Tests/TextLoaderTests.cs

```diff
@@ -937,5 +937,147 @@ public void TestLoadTextWithoutKeyTypeAttribute()
 
             Assert.Equal(expectedCount, data.Schema[1].Type.GetKeyCount());
         }
+
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public void TestLoadTextWithEscapedNewLines(bool useSaved)
+        {
+            var mlContext = new MLContext(seed: 1);
+            var dataPath = GetDataPath("multiline.csv");
+            var baselinePath = GetBaselinePath("TextLoader", "multiline.csv");
+            var options = new TextLoader.Options()
+            {
+                HasHeader = true,
+                Separator = ",",
+                AllowQuoting = true,
+                ReadMultilines = true,
+                Columns = new[]
+                {
+                    new TextLoader.Column("id", DataKind.Int32, 0),
+                    new TextLoader.Column("description", DataKind.String, 1),
+                    new TextLoader.Column("animal", DataKind.String, 2),
+                },
+            };
+
+            var data = mlContext.Data.LoadFromTextFile(dataPath, options);
+            if (useSaved)
+            { 
+                // Check that loading the data view from a text file,
+                // and then saving that data view to another text file, then loading it again
+                // also matches the baseline.
+
+                var savedPath = DeleteOutputPath("saved-multiline.tsv");
+                using (var fs = File.Create(savedPath))
+                    mlContext.Data.SaveAsText(data, fs, separatorChar: '\t');
+
+                options.Separator = "\t";
+                data = mlContext.Data.LoadFromTextFile(savedPath, options);
```

Comment:
I added this to test de TextSaver. (see https://github.com/dotnet/machinelearning/pull/5125/files#r425779339 )